fsw/cfe-core/src/inc/ccsds.h

```diff
@@ -184,6 +184,9 @@ typedef struct
 
 
 
+#ifdef MESSAGE_FORMAT_IS_CUSTOM
+#include "cfe_mission_custom.h"  /* Should resolve/refactor to avoid including mid-file */
+#else
```

Comment:
As the "cfe_mission_custom.h" file needs to exist regardless, couldn't it just be generated to always have the "right" content in it (i.e. the user-selected format) and not have to introduce yet another preprocessor switch?

Reply:
No it doesn't?  Did you try it?  Works exactly like before (remove the file, don't define a format and it's V1, define V2 and it works also...  I confirmed it.  Only needs to exist if you want to redefine.

Reply:
My point is that you've called `generate_config_includefile()` function to create `cfe_mission_custom.h` .... but in the event that there is no config here the file will just be empty.

So why would we want to generate an empty file and then selectively ignore it?  It would be simpler, cleaner to just generate the file with the right content.  No `#ifdef`'s or parallel settings required - just include the generated header file, and it has the right content.

Reply:
Also note that the whole `generate_config_includefile` stuff was originally for backward compatibility with the classic build where things were configured via header file.  While that is still largely the case for CFE because it generally works, there are better methods of doing this in CMake (i.e. how we updated OSAL/osconfig.h recently).  It isn't really a method I'd suggest for new development.

Reply:
I agree 100% with adopting the source selection and CMake techniques implemented in OSAL.  I have suggested that approach multiple times, but it hasn't seemed to have resonated with the group.  For the previous comment - because it works with no impacts on previous configurations.  There's certainly a trade to be made (it can certainly be implemented either way), I'll leave it up to @jwilmot @acudmore since the long term open source solution would help from architecture guidance.

Reply:
Well since we already have both (CMake config file + C headers) in use my recommendation is to favor the CMake method for new development, because it covers both areas (build + code).  

That is, we can control the build and also easily generate a synchronized C header from the CMake variables to influence the actual code.  Using a strictly C header approach is more limiting, as it is not really possible to go the other way, so the information isn't available to CMake for build/source selection.

Reply:
And I continue to agree 100% on improving this to utilize CMake instead of defines in c headers.  It was the solution I was hoping for and was trying to encourage.

Reply:
I do still prefer the core framework implements the basic two formats in the code such that it's covered by cppcheck (I'm not a fan of moving the supported header definitions into sample_defs and generating, but that's a trade we can discuss), but I definitely prefer using CMake to do the configuration work.

Reply:
My point is I don't want users to think they can redefine what version 1 and 2 means (could be implied by moving the definition to sample_defs).  I'd rather control those in the core code.